File: Backend/src/main/java/hu/temalabor/GetFit/model/Activity.java
package hu.temalabor.GetFit.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document("Activity")
public class Activity {
    @Id
    private int _id;                 //azonosito
    private int SportId;            //sport azonosito
    private int UserId;             //felhasznalo azonosito
    private double Time;             //sporttal tolott ido (perc)
    private Date Date;    //sportolas kezdete
    private double Kcal;             //elegetett kaloriak
    private Sport sport;
    private User user;

    public Activity(int id, int SportId, int UserId, double Time, Date Date) {
        super();
        this._id = id;
        this.SportId = SportId;
        this.UserId = UserId;
        this.Time = Time;
        this.Date = Date;
    }


    public void setKcal (){
        Kcal= sport.getKcal()* user.getWeight()*Time;
    }

    public void set_id(int _id) {
        this._id = _id;
    }

    public void setUserId(int userId) {
        UserId = userId;
    }

    public void setKcal(double kcal) {
        Kcal = kcal;
    }

    public void setDate(java.util.Date date) {
        Date = date;
    }

    public void setSportId(int sportId) {
        SportId = sportId;
    }

    public void setTime(double time) {
        Time = time;
    }

    public int get_id() {
        return _id;
    }

    public int getSportId() {
        return SportId;
    }

    public int getUserId() {
        return UserId;
    }

    public double getTime() {
        return Time;
    }

    public java.util.Date getDate() {
        return Date;
    }

    public double getKcal() {
        return Kcal;
    }
}

File: GetFit Flutter App/lib/datas/Activity.dart

class Activity{
    int Id;
    int SportId;
    int UserId;
    double? Time;
    String? Date;
    double? Kcal;
    double Distance;

    Activity( this.Id, this.SportId, this.UserId,this.Time, this.Date,this.Kcal,this.Distance);

    Activity.fromJson(Map<String, dynamic> json)
        :
            Id = json['_id'],
            Time = json['time'],
            Date = json['date'],
            UserId = json['userId'],
            SportId = json['sportId'],
            Kcal = json['kcal'],
            Distance = json['distance'];

    Map<String, dynamic> toJson() => <String, dynamic>{
        '_id': Id,
        'sportId': SportId,
        'userId': UserId,
        'time' : Time,
        'date' : Date,
        'kcal' : Kcal,
        'distance': Distance
    };

}



